jakarta-eclipse/src/org/jakarta/jdt/DiagnosticsCollector.java

```diff
@@ -4,8 +4,10 @@
 
 import org.eclipse.lsp4j.Diagnostic;
 import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.ISourceRange;
 
 
 public interface DiagnosticsCollector {
 	public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnostics);
+	public ISourceRange getRange();
```

Comment:
what is this being used for?

Reply:
It is used for getting the range of diagnostic from collector. The code action param needs position information which can be retrieved from IsoirceRange.